src/vm_pages.rs

```diff
@@ -315,6 +315,58 @@ impl VmRegionList {
         Ok(())
     }
 
+    // Removes the region at [`start`, `end`) of type `region_type`.
+    fn remove(
+        &mut self,
+        start: GuestPageAddr,
+        end: GuestPageAddr,
+        region_type: VmRegionType,
+    ) -> Result<()> {
+        let mut index = self
+            .regions
+            .iter()
+            .enumerate()
+            .find(|(_, r)| r.start <= start && r.end >= end && r.region_type == region_type)
+            .map(|(i, _)| i)
+            .ok_or(Error::InvalidMapRegion)?;
+
+        // Unwrap ok here since we know 'index' is valid.
+        let region = self.regions.get(index).unwrap();
```

Comment:
Instead of doing a `self.regions.get(index)` here, can we get a reference to the region in the `find()` call above?

e.g. ```let (index, region) = self.regions.iter().enumerate().find(...).ok_or(...)?;```

Reply:
Fixed.